addons/stock/models/product_strategy.py

```diff
@@ -26,16 +26,23 @@ class PutAwayStrategy(models.Model):
         'Fixed Locations Per Product', domain=[('product_id', '!=', False)], copy=True)
 
     def putaway_apply(self, product):
+        put_away = self._get_putaway_rule(product)
+        if put_away:
+            return put_away.fixed_location_id
+        return self.env['stock.location']
+
+    @api.multi
```

Comment:
no need for this decorator, by default all methods are multi

Reply:
Doesn't  the guidelines are to always put a decorator?

Reply:
afaik, no